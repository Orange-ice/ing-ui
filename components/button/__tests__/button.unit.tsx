import React from 'react';
import {Button} from '../../index';
import renderer from 'react-test-renderer';
import {fireEvent, render} from '@testing-library/react';

describe('Button 组件测试', () => {
  it('快照测试', () => {
    const dom = renderer.create(<Button effect="primary" variant="outline">Test</Button>);
    expect(dom).toMatchSnapshot();
  });

  it('props 测试', () => {
    const dom = renderer.create(<Button effect="primary" variant="outline">Test</Button>);
    const instance = dom.root;
    expect(instance.findByType(Button).props.effect).toBe('primary');
    expect(instance.findByType(Button).props.variant).toBe('outline');
  });

  it('click 事件', () => {
    const fn = jest.fn();
    const {container} = render(<Button effect="primary" variant="outline" onClick={fn}>Test</Button>);
    fireEvent.click(container.querySelector('button') as HTMLButtonElement);
    expect(fn).toBeCalledTimes(1);
  });
});
