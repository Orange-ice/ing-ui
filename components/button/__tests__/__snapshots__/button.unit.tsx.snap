// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button 组件测试 快照测试 1`] = `
<button
  className="ice-button primary outline"
>
  Test
</button>
`;
